pkg/nuclide-code-format/lib/CodeFormatManager.js

```diff
@@ -62,7 +63,11 @@ class CodeFormatManager {
     }
 
     const provider = matchingProviders[0];
-    if (provider.formatCode != null) {
+    // If formatCode exists, use it if selection isn't empty, or if it's empty
+    // but formatEntireFile doesn't exist.
+    if (provider.formatCode != null &&
+      (!selectionRangeEmpty ||
+        (selectionRangeEmpty && provider.formatEntireFile == null))) {
```

Comment:
But that'd actually be less readable and wouldn't shortcut (it currently does), lol